### PR TITLE
Code refactor on options

### PR DESCRIPTION
**Pull request by huanwei**
1.Describe what this PR did

Refactor code structure on `options`.

2.Does this pull request fix one issue?

During the debug and review of the program, I found the following codes in `mysql-operator/pkg` used/called the struct type of  MySQLOperator Options  defined in `mysql-operator/cmd`, while the latter package should be recognized as the program’s entrypoint. 

```
mysql-operator/pkg/controllers/cluster/controller.go 
mysql-operator/pkg/controllers/cluster/controller_test.go 
mysql-operator/pkg/resources/statefulsets/statefulset.go 
mysql-operator/pkg/resources/statefulsets/statefulset_test.go
```

It looked like the option packages in `mysql-operator/cmd/mysql-operator/app/options` and `mysql-operator/cmd/mysql-agent/app/options` here were not in a good place, so I would like to to tiny refactor the code structure.

3.Describe how you did it

 I did the following changes: 

```
Move `mysql-operator/cmd/mysql-operator/app/options` to `mysql-operator/pkg/options/operator`
Move `mysql-operator/cmd/mysql-agent/app/options` to `mysql-operator/pkg/options/agent`
Rename struct  type `MySQLOperatorServer` to `MySQLOperatorOpts`, with respect to the struct type of `MysqlAgentOpts`
Adjust the usage of struct type MySQLOperatorOpts passed to the codes in `mysql-operator/pkg` and `mysql-operator/cmd`.
```

4.Describe how to verify it

Compile the code, build the docker images, and verify the operator, agent and cluster on Kubernetes.
```
[root@k8s-master ~]# kubectl get pods -n mysql-operator
NAME                              READY     STATUS    RESTARTS   AGE
mysql-cluster-with-3-replicas-0   2/2       Running   0          7d
mysql-cluster-with-3-replicas-1   2/2       Running   0          7d
mysql-cluster-with-3-replicas-2   2/2       Running   0          7d
mysql-operator-78b7d9bfcf-nfcp8   1/1       Running   0          16d
```

### CODE REVIEW

**Comment by huanwei** [comment]
@owainlewis 
I have submitted the OCA this morning. Pls kindly check it. Thank you.

**Comment by delabassee** [comment]
@owainlewis FYI, @huanwei OCA has been reviewed and approved

File: cmd/mysql-operator/app/mysql_operator.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
 
-	options "github.com/oracle/mysql-operator/cmd/mysql-operator/app/options"
+	opratoropts "github.com/oracle/mysql-operator/pkg/options/operator"
```
Comment (owainlewis) [inline cmd/mysql-operator/app/mysql_operator.go]: Minor typo here (opratoropts => operatoropts)

File: cmd/mysql-operator/app/mysql_operator.go [hunk 2]
```diff
@@ -45,15 +45,15 @@ const (
 
 // resyncPeriod computes the time interval a shared informer waits before
 // resyncing with the api server.
-func resyncPeriod(s *options.MySQLOperatorServer) func() time.Duration {
+func resyncPeriod(s *opratoropts.MySQLOperatorOpts) func() time.Duration {
```
Comment (owainlewis) [inline cmd/mysql-operator/app/mysql_operator.go]: Minor typo here (opratoropts => operatoropts)

File: cmd/mysql-operator/app/mysql_operator.go [hunk 3]
```diff
@@ -45,15 +45,15 @@ const (
...
 	return func() time.Duration {
 		factor := rand.Float64() + 1
 		return time.Duration(float64(s.MinResyncPeriod.Nanoseconds()) * factor)
 	}
 }
 
 // Run starts the mysql-operator controllers. This should never exit.
-func Run(s *options.MySQLOperatorServer) error {
+func Run(s *opratoropts.MySQLOperatorOpts) error {
```
Comment (owainlewis) [inline cmd/mysql-operator/app/mysql_operator.go]: Minor typo here (opratoropts => operatoropts)

**Comment by owainlewis** [comment]
Thanks for this contribution @huanwei the changes seem reasonable to me in terms of code organisation.

File: cmd/mysql-operator/app/mysql_operator.go (hunk 1 above)
Comment (huanwei) [reply cmd/mysql-operator/app/mysql_operator.go]: Thanks for the review.

File: cmd/mysql-operator/app/mysql_operator.go (hunk 3 above)
Comment (huanwei) [reply cmd/mysql-operator/app/mysql_operator.go]: Thanks for the review.

File: cmd/mysql-operator/app/mysql_operator.go (hunk 2 above)
Comment (huanwei) [reply cmd/mysql-operator/app/mysql_operator.go]: Thanks for the review.

File: cmd/mysql-operator/app/mysql_operator.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
 
-	options "github.com/oracle/mysql-operator/cmd/mysql-operator/app/options"
+	operatoropts "github.com/oracle/mysql-operator/pkg/options/operator"
```
Comment (huanwei) [inline cmd/mysql-operator/app/mysql_operator.go]: @owainlewis Pls review the change. Thanks.

File: cmd/mysql-operator/app/mysql_operator.go [hunk 5]
```diff
@@ -45,15 +45,15 @@ const (
 
 // resyncPeriod computes the time interval a shared informer waits before
 // resyncing with the api server.
-func resyncPeriod(s *options.MySQLOperatorServer) func() time.Duration {
+func resyncPeriod(s *operatoropts.MySQLOperatorOpts) func() time.Duration {
```
Comment (huanwei) [inline cmd/mysql-operator/app/mysql_operator.go]: @owainlewis Pls review the change. Thanks.

File: cmd/mysql-operator/app/mysql_operator.go [hunk 6]
```diff
@@ -45,15 +45,15 @@ const (
...
 	return func() time.Duration {
 		factor := rand.Float64() + 1
 		return time.Duration(float64(s.MinResyncPeriod.Nanoseconds()) * factor)
 	}
 }
 
 // Run starts the mysql-operator controllers. This should never exit.
-func Run(s *options.MySQLOperatorServer) error {
+func Run(s *operatoropts.MySQLOperatorOpts) error {
```
Comment (huanwei) [inline cmd/mysql-operator/app/mysql_operator.go]: @owainlewis Pls review the change. Thanks.

**Pull request merged**